bin/vagrant

```diff
@@ -78,6 +78,12 @@ if argv.include?("--debug-timestamp")
   ENV["VAGRANT_LOG_TIMESTAMP"] = "1"
 end
 
+# Setting to enable/disable showing progress bars
+if argv.include?("--quiet-progress")
+  argv.delete("--quiet-progress")
+  ENV["VAGRANT_QUIET_PROGRESS"] = "1"
```

Comment:
With having a dedicated UI subclass, we won't need to worry about tracking in an environment variable. Instead, just use the option to instantiate the new subclass here: https://github.com/hashicorp/vagrant/blob/e5d4243aecde7526176d16441f2252f744614885/bin/vagrant#L118